cli/src/opts.rs

```diff
@@ -48,11 +48,6 @@ pub struct ServeArgs {
     #[arg(long = "addr")]
     socket_addr: Option<SocketAddr>,
 
-    /// Whether to profile the wasm guest. Takes an optional directory to save
-    /// per-request profiles to
-    #[arg(long, default_missing_value = "guest-profiles", num_args=0..=1, require_equals=true)]
-    profile_guest: Option<PathBuf>,
-
```

Comment:
Note: This would be a breaking change as we are removing a flag, we'd need to version accordingly